CONTRIBUTING.md

```diff
@@ -502,6 +510,45 @@ kubectl -n crossplane-system scale deploy crossplane --replicas=0
 make run
 ```
 
+## Code Review Process
+
+All Pull Requests (PR), whether written by a Crossplane maintainer or a
+community member, must go through code review. This ensures that the code is
+correct, maintainable, and secure.
+
+We encourage anyone in the community to conduct a code review on a PR,
+however, we require the following approvals before merging a PR:
+- At least one approval from [Reviewers] or [Code Owners]
+- At least one approval from [Maintainers]
+
+When opening a PR, GitHub will automatically assign the
+[crossplane-reviewers] and [crossplane-maintainers] teams, as well as
+any [Code Owners]. The PR author may also add any additional reviewers they
+might want to have a look at the PR.
+
+If the PR author has a specific code reviewer in mind, we encourage them to
+remove the auto-assigned reviewers and directly add the individuals they would
+like to review the code.
+
+[Maintainers] will be the last to review a PR, after the initial review from
+[Reviewers] and are the only ones that can merge a PR.
+
+The PR author should notify the [crossplane-maintainers] team or the
+individual maintainer assigned to the PR, when the PR is ready for their final
+review.
+
+### Reviewers vs Maintainers
+
+[Reviewers] (or [Code Owners]) are there to ensure the code is functionally
+correct, written according to the [Coding Style](#coding-style), and
+sufficiently covered by tests. They take care of basic PR hygience by ensuring
+the PR author has signed the DCO, and all CI jobs are passing before giving
+their approval.
+
+[Maintainers] will conduct a review focusing on ensuring APIs are
+idiomatically aligned with the existing Crossplane codebase and does not steer
+the project in the wrong direction.
```

Comment:
This is not untrue, but is not exhaustive. For example there's plenty of other factors besides API design that maintainers would consider, and I think we maintainers would be more than happy if reviewers wanted to jump in with feedback on more 'advanced' topics (API design etc).

I think the key difference between a reviewer and a maintainer is experience; there's a low bar (essentially just willingness to help) to become a reviewer so we don't _expect_ that reviewers will have the context about the project to review for idiomatic approaches etc, while we do expect that from maintainers.

Reply:
@negz you mention there's a low bar to become a reviewer, but at the same time, we don't set the bar low. Someone can look at the project and think it's just too complicated to get involved in doing reviews. By clarifying the expectation of the reviewer and maintainer and sharing the responsibility to ensure high quality it could make it more palatable for folks to do reviews.

I'd like to encourage we keep this section and maybe expand on it to make it more true.